Branches/Stable/Behaviors/obj_Hauler.iss

```diff
@@ -1008,11 +1008,11 @@ objectdef obj_Hauler
 						wait 20
 					}
 				}
-				if ${Ship.CargoFreeSpace} < 1000
-				{
-					UI:UpdateConsole["DEBUG: obj_Hauler.LootEntity: Ship Cargo Free Space: ${Ship.CargoFreeSpace} < ${Ship.CargoMinimumFreeSpace}"]
-					break
-				}
+				; if ${Ship.CargoFreeSpace} < 1000

```

Comment:
This is required; if it's not here, when the free space drops to 999.99, and you're ice mining, the loop will get stuck trying to move 1k blocks of ore.  Free space values are not always accurate.
Instead, replace the code you commented out with a call to HaulerFull -- then it'll just make the 1k check when it's ice mining.